ui/server/src/main/scala/pl/touk/nussknacker/ui/component/ComponentService.scala

```diff
@@ -139,6 +162,23 @@ class DefaultComponentService private (config: Config,
     }
   }
 
+  override def getComponentProcesses(componentId: ComponentId)(implicit user: LoggedUser): Future[XError[List[ComponentProcess]]] =
+    componentsIdStorage
+      .get(componentId)
+      .map(getComponentProcesses(_).map(Right(_)))
+      .getOrElse(Future(Left(ComponentNotFoundError(componentId))))
+
+  private def getComponentProcesses(componentId: String)(implicit user: LoggedUser): Future[List[ComponentProcess]] = {
+    val userCategories = categoryService.getUserCategories(user)
+    fetchingProcessRepository
+      .fetchProcesses[DisplayableProcess](None, None, None, categories = Some(userCategories), None)
```

Comment:
I wonder, do we want this logic user->categories->repo in ProcessService? Not in this change probably :P but still...

Reply:
Changed. In next PR I will do the same think with loadSubprocesses..